go/docs/port_detection.md

```diff
@@ -0,0 +1,287 @@
+If one or more ports are found in one step, the rest are skipped. (E.g - if a port is found in the docker file, Alizer will not search and analyze 
+any docker-compose file and the source code).
+
+It is also possible to customize the way Alizer searches for ports by defining the detection strategies to use and their order.
+For example, by using a source and docker strategies, Alizer analyzes the source code and then the docker file, if any. Or docker and compose, to avoid analyzing the source.
+
+If no port is found an empty list is returned.
+
+### Port detection with a docker-compose file
+
+The port detection in a `[compose|docker-compose].[yml|yaml]` file targets the `expose` and `ports` fields of the service related to the component, if any. 
+To search for the specific component, Alizer look at the `build` field.
+
+Example of `expose` - the result is [3000,8000]
+```
+services:
+  web:
+    ...
+    expose:
+      - "3000"
+      - "8000"
+  myapp2:
+    ...
+    expose:
+      - "5000"
+```
+
+Example of short syntax `ports` (`[HOST:]CONTAINER[/PROTOCOL]`) - the result is [3000,3005,8000,8081,8002,6060]
+```
+services:
+  web:
+    ...
+  ports:
+    - "3000"                             # container port (3000), assigned to random host port
+    - "8000:8000"                        # container port (8000), assigned to given host port (8000)
+    - "127.0.0.1:8002:8002"              # container port (8002), assigned to given host port (8002) and bind to 127.0.0.1
+    - "6060:6060/udp"                    # container port (6060) restricted to UDP protocol, assigned to given host (6060)
+  myapp2:
+    ...
+    expose:
+      - "5000"
+```
+
+Example of long syntax `ports` - the result is [6060]
+```
+services:
+  web:
+    ...
+  ports:
+    - target: 6060
+      host_ip: 127.0.0.1
+      published: 6060
+      protocol: udp
+      mode: host
+  myapp2:
+    ...
+    expose:
+      - "5000"
+```
+
+### Port detection with frameworks
+### Java Frameworks
+
+#### Micronaut
+
+Alizer checks if the environment variable MICRONAUT_SERVER_PORT is set. If not or it does not contain a valid port value, it searches for the `application.[yml|yaml]` file in `src/main/resources` folder and verify if a port is set.
+
+The known schema is:
+```
+micronaut:
+    server:
+        port: <port>
+```
+
+#### OpenLiberty
+
+Alizer searches for the `server.xml` file within the root or `src/main/liberty/config` folder and verify if a port is set.
+
+The known schema is:
+```
+<server>
+    <httpEndpoint id="defaultHttpEndpoint"
+        httpPort="<port>"
+        httpsPort="<port>" />
+</server>
+```
+
+#### Quarkus 
+
+Alizer follows the default behavior of Quarkus which reads configuration properties from multiple sources (by descending ordinal).
+
+1) It checks if the environment variable QUARKUS_HTTP_PORT is set
+2) It checks for the `QUARKUS_HTTP_PORT` within the `.env` file, if any, located in the root
```

Comment:
Quarkus may use QUARKUS_HTTP_SSL_PORT or QUARKUS_HTTP_PORT or both based on QUARKUS_HTTP_INSECURE_REQUEST.
See https://quarkus.io/guides/all-config and use https as a filter.